eth/tracers/internal/tracetest/testdata/prestate_tracer_with_diff_mode/suicide.json

```diff
@@ -0,0 +1,109 @@
+        "storage": {
+          "0x0000000000000000000000000000000000000000000000000000000000000000": "0x000000000000000000000000d3cda913deb6f67967b99d67acdfa1712c293601",
+          "0x0000000000000000000000000000000000000000000000000000000000000001": "0x0000000000000000000000ff30c9e568f133adce1f1ea91e189613223fc461b9"
+        }
+      },
+      "0x2a65aca4d5fc5b5c859090a6c34d164135398226": {
+        "balance": "0x326601cc6cf364f6b9",
+        "nonce": 12122,
+        "code": "0x"
+      },
+      "0xd3cda913deb6f67967b99d67acdfa1712c293601": {
+        "balance": "0x1ff0509d9d6821e26",
+        "nonce": 138,
+        "code": "0x"
+      }
+    },
+    "post": {
```

Comment:
Curious about this 'suicide.json' test -- did any account here do a selfdestruct? If so, it doesn't really show on the post-state?

Reply:
yes, the contract 0x2861bf89b6c640c79040d357c1e9513693ef5d3f is selfdestructed, and only in the `pre` stage . actually it's this transaction [0x0de8b6a364b7ab3425069cd5f03acaa3a6fc1e560c61ae7cadab0627f87178d0](https://etherscan.io/tx/0x0de8b6a364b7ab3425069cd5f03acaa3a6fc1e560c61ae7cadab0627f87178d0)

Reply:
What happens with contracts that are not modified during the execution (but still read / accessed )? Are they present in the post-state too, but empty?

Reply:
In the current implementation, If his balance/storage were modified, then exists in post-state, if only read/accessed, then this account will be in pre or post-state.

But, I think we can add another option to keep them all in pre-post.

Reply:
>  selfdestructed, and only in the pre stage .

>  if only read/accessed, then this account will be in pre or post-state.

Not sure I follow. How does one distinguish between "not in post-state because nothing changed" versus "not in post-state, because it was destructed" ?

Reply:
I think that's a typo a should read "if only read/accessed, then this account will **NOT** be in pre or post-state."